core/src/sampling/RandomSampler.cpp

```diff
@@ -139,69 +139,27 @@ void RandomSampler::draw(std::vector<size_t>& result,
                          size_t max,
                          const std::set<size_t>& skip,
                          size_t num_samples) {
-  if (num_samples < max / 2) {
```

Comment:
I'm not sure we should remove the `draw_simple` algorithm. I think the motivation is that for small numbers of samples, the simple strategy can be substantially faster. Would it be possible to follow the same strategy as in the ranger commit here, where we keep the two sampling strategies, and decrease the threshold to `max / 10`?

Reply:
Of course. Will do.